internal/metrics/sample_counter_test.go

```diff
@@ -5,19 +5,92 @@
 
 package metrics_test
 
-import "github.com/envoyproxy/gateway/internal/metrics"
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"os"
+	"testing"
 
-var (
-	irUpdates = metrics.NewCounter(
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+
+	"github.com/envoyproxy/gateway/internal/metrics"
+)
+
+func TestCounter(t *testing.T) {
+
+	writer := os.Stdout
+	if true {
```

Comment:
@ShyunnY why always true?

Reply:
Since the metric carries a timestamp, it needs to be written before each test is executed. (So explicitly make it true)

Reply:
if always true, you shuld just remove this line?

Reply:
BTW, these file will be updated every time you run the test, so you'd better improve it.

Reply:
Thanks for the suggestions, I will improve this soon, thank you again!